src/interpreter.rs

```diff
@@ -39,8 +43,8 @@ struct Word<T> {
 }
 
 impl<T> Word<T> {
-    fn eval_within(&self, i: &mut Interpreter<T>) {
-        (self.entry)(i)
+    fn eval_within(&self, mut i: &mut Interpreter<T>) {
+      (self.entry)(&mut i)
     }
```

Comment:
But it's the interpreter that can mutate, not its reference… I don't intend to make `i` point to another interpreter, only to change the interpreter's state